lib/lightstep/span.rb

```diff
@@ -1,16 +1,24 @@
 require 'concurrent'
+require 'forwardable'
+require 'lightstep/span_context'
 
 module LightStep
   # Span represents an OpenTracer Span
   #
   # See http://www.opentracing.io for more information.
   class Span
+    extend Forwardable
+
     # Part of the OpenTracing API
     attr_writer :operation_name
 
     # Internal use only
     # @private
-    attr_reader :guid, :trace_guid, :start_micros, :end_micros, :baggage, :tags, :operation_name
+    attr_reader :start_micros, :end_micros, :baggage, :tags, :operation_name, :span_context
+
+    def_delegator :span_context, :id
```

Comment:
I have a preference to remove `:id` and `:trace_id` from `Span` per se. I get that it saves typing, but history has shown that there's benefit to forcing people to be explicit and really understand what they're doing (by descending into the context).

I don't care as much about `baggage`...